components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1261,9 +1408,19 @@ public void addIdP(IdentityProvider identityProvider, String tenantDomain)
                     + identityProvider.getIdentityProviderName() + " for tenant " + tenantDomain;
             throw new IdentityProviderManagementException(msg);
         }
+        StringBuilder idpName = new StringBuilder("");
+        StringBuilder metadata = new StringBuilder("");
+
+        handleMetadta(identityProvider, idpName, metadata);
 
         validateIdPEntityId(identityProvider.getFederatedAuthenticatorConfigs(), tenantId, tenantDomain);
+        if (
+                idpName != null && idpName.toString().length() > 0 &&
```

Comment:
Can idpName and metadata be null since you are initializing them in lines #1411 and #1412?